src/chunk-optimization.ts

```diff
@@ -0,0 +1,144 @@
+import Chunk from './Chunk';
+import ExternalModule from './ExternalModule';
+import { OutputOptions } from './rollup';
+
+/*
+ * Given a chunk list, perform optimizations on that chunk list
+ * to reduce the mumber of chunks. Mutates the chunks array.
+ *
+ * Manual chunks (with chunk.chunkAlias already set) are preserved
+ * Entry points are carefully preserved as well
+ *
+ */
+export function optimizeChunks(
+	chunks: Chunk[],
+	options: OutputOptions,
+	CHUNK_GROUPING_SIZE = 5000
+): Chunk[] {
+	const chunkSize = new Map<Chunk, number>();
+	function getChunkSize(chunk: Chunk) {
+		let size = chunkSize.get(chunk);
+		if (size) return size;
+		size = chunk.getRenderedSourceLength();
+		chunkSize.set(chunk, size);
+		return size;
+	}
+
+	for (let i = 0; i < chunks.length; i++) {
```

Comment:
The variables `i` and `j` are used over a span of more than 120 lines. It would really help readability if they had names that tell a little better what they stand for.